atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -303,7 +325,24 @@ private HikariDataSource getDataSourcePool() {
         HikariDataSource dataSourcePool;
 
         try {
-            dataSourcePool = new HikariDataSource(hikariConfig);
+            try {
+                dataSourcePool = new HikariDataSource(hikariConfig);
+            } catch (IllegalArgumentException e) {
+                // allow multiple pools on same JVM (they need unique names / endpoints)
+                if (e.getMessage().contains("A metric named")) {
+                    String poolName = connConfig.getConnectionPoolName();
+                    String newPoolName =
+                            poolName + "-" + ThreadLocalRandom.current().nextInt();
+                    hikariConfig.setPoolName(newPoolName);
+                    dataSourcePool = new HikariDataSource(hikariConfig);
+                    log.error(
```

Comment:
can we do `log.warn` instead? Some internal users have alerts on these and I don't think we need to file a P0 here

Reply:
Done.